src/components/sidebar-content-box/SidebarContentBox.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { FC } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+
+import AppButton from '~/components/app-button/AppButton'
+import AllContentModal from '~/components/all-content-modal/AllContentModal'
+import FileComponent from '~/components/file-component/FileComponent'
+import LinkComponent from '~/components/link-component/LinkComponent'
+
+import { useModalContext } from '~/context/modal-context'
+import { spliceSx } from '~/utils/helper-functions'
+import { SizeEnum, ButtonVariantEnum, Link, File } from '~/types'
+
+import { styles } from '~/components/sidebar-content-box/SidebarContentBox.styles'
+
+interface SidebarContentBoxProps {
+  Icon: React.ComponentType
+  name: string
+  content?: Link[] | File[]
+}
+
+type SidebarImageGridModifications = {
+  compactMode: boolean
+}
+
+const SidebarContentBox: FC<SidebarContentBoxProps> = ({
+  Icon,
+  name,
+  content,
+  children
+}) => {
+  const maxElemToShow = 3
+
+  const { t } = useTranslation()
+  const { openModal } = useModalContext()
+
+  const seeAll = () => {
+    const childrenCopy = React.Children.map(children, (child) => {
+      if (
+        React.isValidElement(child) &&
+        (child.type as React.FunctionComponent).name === 'SidebarImageGrid'
+      ) {
+        return React.cloneElement(child, {
+          compactMode: false
+        } as SidebarImageGridModifications)
+      }
+      return child
+    })
```

Comment:
It looks very weird.

Reply:
> It looks very weird.

In this component I need to change the children so that it have props compact mode if this component is an SidebarImageGrid.
I can write it a little better, but this option is the only one I can think of.

```ts
const updatedChildren = React.Children.map(children, (child) => {
  if (isValidElement(child) && child.type === SidebarImageGrid) {
    return cloneElement(child, { compactMode: false } as ModifiedChildren)
  }
  return child
})